business/views.py

```diff
@@ -11,12 +12,25 @@ def sponsor(request):
 
 
 def jobs(request):
-    offers_objs = Offer.objects.filter(
-        Q(ends_at__isnull=True) | Q(ends_at__gt=timezone.now()),
-        is_visible=True,
-        starts_at__lte=timezone.now(),
-    ).order_by("-created_at")
-    return render(request, "jobs.html", {"offers": offers_objs})
+    type = request.GET.get("type", "")
+    if type == "":
```

Comment:
Much simpler to...
```python
offers_objs = Offer.objects.filter(
    Q(ends_at__isnull=True) | Q(ends_at__gt=timezone.now()),
    is_visible=True,
    starts_at__lte=timezone.now(),
).order_by("-created_at")
if type is not None:
    offers_objs = offers_objs.filter(type=type)
```

Or this would also do the trick.
```python
offer_filters = Q(
    Q(ends_at__isnull=True) | Q(ends_at__gt=timezone.now()),
    is_visible=True,
    starts_at__lte=timezone.now()
)
if type is not None:
    offer_filters &= Q(type=type)
offers_objs = Offer.objects.filter(offer_filters).order_by("-created_at")
```

Note the queryset won't be evaluated (aka no DB call) until the `render` call as this is lazy evaluation (no computing until results are needed).